docs/topics/multiplatform/multiplatform-hierarchy.md

```diff
@@ -1,111 +1,340 @@
+
+> In this example, `apple` and `native` source sets compile only to `iosArm64` and `iosSimulatorArm64` targets.
+> Therefore, despite their names, they have access to the full iOS API.
+> This might be counter-intuitive for source sets like `native`, as you may expect that only APIs available on all
+> native targets are accessible in this source set. This behavior may change in the future.
+>
+{type="note"}
+
+### Adjust resulting hierarchy
+
+If necessary, you can further configure the resulting hierarchy manually [using the `dependsOn` relation](#manual-configuration).
+To do that, apply the `by getting` construction for the source sets created with `targetHierarchy.default()`.
+
+Consider this example of a project with a source set shared between the `jvm` and `native` targets only:
+
+```kotlin
+@OptIn(ExperimentalKotlinGradlePluginApi::class)
+kotlin {
+    // Enable the default target hierarchy:
+    targetHierarchy.default()
+
+    jvm()
+    iosArm64()
+    // the rest of the necessary targets...
+
+    sourceSets {
+        val commonMain by getting
+
+        val jvmAndNativeMain by creating {
+            dependsOn(commonMain)
+        }
+
+        val nativeMain by getting {
+            dependsOn(jvmAndNativeMain)
+        }
+
+        val jvmMain by getting {
+            dependsOn(jvmAndNativeMain)
+        }
+    }
+}
 ```
-{initial-collapse-state="collapsed"}
 
-Instead of doing this, you can create a hierarchical structure
-with [target shortcuts](multiplatform-share-on-platforms.md#use-target-shortcuts)
-available for typical multi-target scenarios, or you can manually declare and connect the source sets. For example, you
-can create two iOS targets and a shared source set with the `ios()` shortcut:
+Removing `dependsOn` relations automatically created by `targetHierarchy.default()` could be cumbersome.
+If it's necessary, use entirely [manual configuration](#manual-configuration) instead of calling default hierarchy.
+
+> We're currently working on API for creating your own target hierarchies. It should be useful for projects
+> which hierarchy configurations differ a lot from the default template.
+>
+> At the moment, this API is not ready yet. But if you're eager to try it,
+> look into the `targetHierarchy.custom { ... }` block and the declaration of `targetHierarchy.default()` as an example.
+> Keep in mind that this API is still in development. It might be under-tested, and can change dramatically in further releases.
+> 
+{type="tip"}
+
+#### See the full hierarchy template {initial-collapse-state="collapsed"}
+
+When you declare the targets to which your project compiles,
+the plugin picks shared source sets based on the specified targets from the template and creates them in your project.
+
+![Default target hierarchy](full-template-hierarchy.svg)
+
+> Here we describe the production part of the project only, omitting the suffix `Main`
```

Comment:
This example shows the production part of the project, omitting the `Main` suffix.
...